course7/homeworks/anton/ls/main.cc

```diff
@@ -0,0 +1,135 @@
+#include <boost/filesystem.hpp>
+#include <boost/program_options.hpp>
+#include <algorithm>
+#include <iomanip>
+#include <iostream>
+
+namespace fs = boost::filesystem;
+namespace po = boost::program_options;
+
+struct configuration
+{
+    std::string file_path = ".";
+    bool show_help = false;
+    bool show_tree = false;
+    bool show_long = false;
+};
+
+configuration parse_args(int argc, char *argv[]);
+void print_file_tree(const fs::path& dir, const configuration& config, const std::string& filler = "");
+size_t get_file_size(const fs::path&);
+
+int main(int argc, char *argv[])
+{
+    try
+    {
+        const configuration& config = parse_args(argc, argv);
+
+        if (!config.show_help)
+        {
+            fs::path path(config.file_path);
+            print_file_tree(path, config);
+        }
+
+        return 0;
+    }
+    catch(std::exception& ex)
+    {
+        std::cout << "\033[1;31m" << ex.what() << "\033[0m" << std::endl;
+        return 1;
+    }
+}
+
+configuration parse_args(int argc, char *argv[])
+{
+    configuration config;
+    po::positional_options_description po;
+    po.add("file", 1);
+
+    po::options_description desc("Allowed options");
+    desc.add_options()
+        ("help,h", po::bool_switch(&config.show_help), "show help")
+        ("tree,t", po::bool_switch(&config.show_tree), "show tree")
+        ("long,l", po::bool_switch(&config.show_long), "show long")
+        ("file", po::value<std::string>(&config.file_path), "file path")
+    ;
+
+    auto print_help = [&]
```

Comment:
Oh this could be just a simple function! no need for this.

Reply:
changed to function. but i don't like both variants. Possibly allowing the code duplication could be even better.